frame/asset-rate/src/lib.rs

```diff
@@ -0,0 +1,239 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Asset Rate Pallet
```

Comment:
probably would be better with a plural form - `assets-rate`, for consistency with existing `assets`.

Reply:
could be also `assets-price`

Reply:
No, nouns usually take singular form when used as adjectives. A "bike lane" is a "lane for bikes", we don't say "bikes lane".